proposals/heps/0002-YAML-converter.md

```diff
@@ -0,0 +1,471 @@
+# Meta
+[meta]: #meta
+- Name: YAML Code Converter
+- Start Date: (fill in today's date: 2025-04-29)
+- Update date (optional): (fill in today's date: YYYY-MM-DD)
+- Author(s): [@elliotgunton](https://github.com/elliotgunton)
+- Supersedes: N/A
+
+# Table of Contents
+[table-of-contents]: #table-of-contents
+- [Meta](#meta)
+- [Table of Contents](#table-of-contents)
+- [Overview](#overview)
+- [Definitions](#definitions)
+- [Motivation](#motivation)
+- [Proposal](#proposal)
+  - [Terminology](#terminology)
+  - [Code Examples](#code-examples)
+- [Implementation](#implementation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Prior Art](#prior-art)
+
+# Overview
+[overview]: #overview
+
+This HEP introduces a way to generate boilerplate Hera code from existing YAML Workflows, allowing existing Argo Workflows users to get their adoption of Hera kick started.
+
+# Definitions
+[definitions]: #definitions
+
+Make a list of the definitions that may be useful for those reviewing. Include phrases and words that Hera users or other interested parties may not be familiar with.
```

Comment:
just FYI, I've been holding off on reviewing because this PR looks incomplete (here for example)

Reply:
The PR is ready for review, these two definitions/terminology sections are if there's anything you want to point out as unclear that I can then add. Otherwise I can delete these sections.